data_validation/__main__.py

```diff
@@ -252,53 +255,29 @@ def build_config_from_args(args: Namespace, config_manager: ConfigManager):
-            config_manager.append_aggregates(get_aggregate_config(args, config_manager))
-
-        # For custom-query row command
-        if args.custom_query_type == consts.ROW_VALIDATION.lower():
-            # Append Comparison fields
-            if args.comparison_fields is not None:
-                comparison_fields = cli_tools.get_arg_list(
-                    args.comparison_fields, default_value=[]
-                )
-
-                # As per #1190, add rstrip for Teradata string comparison fields
-                if (
-                    config_manager.source_client.name == "teradata"
-                    or config_manager.target_client.name == "teradata"
-                ):
-                    comparison_fields = config_manager.add_rstrip_to_comp_fields(
-                        comparison_fields
-                    )
-
-                config_manager.append_comparison_fields(
-                    config_manager.build_config_comparison_fields(comparison_fields)
-                )
-
-            # Append calculated fields: --hash/--concat
-            config_manager.append_calculated_fields(
-                get_calculated_config(args, config_manager)
-            )
-
-            # Append primary_keys
-            primary_keys = cli_tools.get_arg_list(args.primary_keys)
-            config_manager.append_primary_keys(
-                config_manager.build_column_configs(primary_keys)
-            )
-
-    # Append COLUMN_VALIDATION configs
-    if config_manager.validation_type == consts.COLUMN_VALIDATION:
+    # Append COLUMN_VALIDATION configs, including custom-query column validation
+    if (
+        config_manager.validation_type == consts.COLUMN_VALIDATION
+        or config_manager.validation_type == consts.CUSTOM_QUERY
+        and args.custom_query_type == consts.COLUMN_VALIDATION.lower()
+    ):
         config_manager.append_aggregates(get_aggregate_config(args, config_manager))
-        if args.grouped_columns is not None:
+        if (
+            config_manager.validation_type == consts.COLUMN_VALIDATION
+            and args.grouped_columns
+            is not None  # grouped_columns not supported in custom queries - at least now.
+        ):
             grouped_columns = cli_tools.get_arg_list(args.grouped_columns)
             config_manager.append_query_groups(
                 config_manager.build_column_configs(grouped_columns)
             )
 
-    # Append ROW_VALIDATION configs
-    if config_manager.validation_type == consts.ROW_VALIDATION:
+    # Append ROW_VALIDATION configs, including custom-query row validation
+    if (
+        config_manager.validation_type == consts.ROW_VALIDATION
+        or config_manager.validation_type == consts.CUSTOM_QUERY
+        and args.custom_query_type == consts.ROW_VALIDATION.lower()
+    ):
         # Append calculated fields: --hash/--concat
         config_manager.append_calculated_fields(
```

Comment:
Are there any effects of appending the calculated fields _before_ comparison fields? Before this PR, the order was reversed for custom query.

Reply:
I saw that as well, but forgot to double check it. Now I have. Calculated_fields (hash or concat) and comparison_fields are mutually exclusive - in the argparse, so only one of these if statements is ever triggered.